torch/lib/THC/generic/THCTensorMathReduce.cu

```diff
@@ -368,11 +368,11 @@ THCTensor_(median)(THCState *state,
                    THCTensor *values,
                    THCudaLongTensor *indices,
                    THCTensor *self,
-                   long dimension,
+                   int dimension,
                    int keepdim) {
   THCAssertSameGPU(THCTensor_(checkGPU)(state, 1, self));
 
-  long t_size_dim, k;
+  int t_size_dim, k;
```

Comment:
Shouldn't this be `int64_t`? In line 377 you use it in `THCTensor_(size)`, which returns `int64_t`, right?